gh-connector/src/main/java/de/tum/gh_connector/service/GHConnectorService.java

```diff
@@ -0,0 +1,127 @@
+package de.tum.gh_connector.service;
+
+import de.tum.gh_connector.client.GHRestClient;
+import de.tum.gh_connector.dto.ContentResponseItem;
+import de.tum.gh_connector.dto.GHConnectorResponse;
+import de.tum.gh_connector.dto.WorkflowFile;
+import feign.FeignException;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class GHConnectorService {
+
+    private final GHRestClient ghRestClient;
+
+    public GHConnectorResponse analyzeRepo(String repoUri) {
+
+        System.out.println("got called with: " + repoUri);
+
+        String contentPath;
+        try {
+            contentPath = constructGHApiContentPath(repoUri);
+            assertWorkflowDirAccess(contentPath);
+
+            return crawlWorkflows(contentPath);
+        } catch (IllegalArgumentException e) {
+            return constructError("There was an error while working with the provided URL: " + e.getMessage());
+        }
+
+//        return constructError("not implemented yet");
+    }
+
+    private GHConnectorResponse crawlWorkflows(String contentPath) {
+        List<WorkflowFile> workflowFiles = new ArrayList<>();
+
+        try {
+            crawlWorkflows(".github/workflows", contentPath, workflowFiles);
+        } catch (Exception e) {
+            //TODO
+            e.printStackTrace();
+        }
+
+        return GHConnectorResponse.builder()
+                .status("200")
+                .files(workflowFiles)
+                .build();
+    }
```

Comment:
I don't really like having a http status in the payload. Why not use the actual http status code of the response?